### ISSUE TITLE
Why am I getting a discrepency between SSDLite Scores and the Full Probability Vector?

### ISSUE THREAD

**Issue by Aneesh-Sandhir** [pytorch/vision#9196]
I am noticitng a slight discrepency between the scores output by the SSDLite model and the Full Probability Vector you get from feeding the features extracted from the backbone through the model head. While the difference is slight, around .004, I find the behavior peculiar and cant find an explanation. Please see the code below:

```
import torch
from torchvision.models.detection import ssdlite320_mobilenet_v3_large
from torchvision.transforms import functional as F
from PIL import Image
import requests 

model = ssdlite320_mobilenet_v3_large(weights=True)
model.eval();

model_categories_url = 'https://raw.githubusercontent.com/levan92/coco-classes-mapping/refs/heads/master/coco91.names'
model_categories = requests.get(model_categories_url).text.split('\n')
model_categories.insert(0, '')

image_url = 'https://storage.googleapis.com/download.tensorflow.org/example_images/YellowLabradorLooking_new.jpg'
image_label = 'dog'
img = Image.open(requests.get(image_url, stream = True).raw)
img_tensor = F.to_tensor(img)
img_tensor = F.resize(img_tensor, [320, 320])
img_tensor = img_tensor.unsqueeze(0)

with torch.no_grad():
    # 1. Pass through backbone and head
    backbone_output = model.backbone(img_tensor)
    
    #2 Convert OrderedDict to list of feature maps
    features = list(backbone_output.values())
    head_outputs = model.head(features)

    # 3. Compute class logits (before NMS)
    class_logits = head_outputs['cls_logits']  # shape: [batch_size, num_anchors, num_classes]
    bbox_regression = head_outputs['bbox_regression']

    # 4. Apply softmax to get probabilities
    class_probs = torch.softmax(class_logits, dim=-1)  # shape: [1, num_anchors, num_classes]

class_index = model_categories.index(image_label)
class_detections = class_probs[0, (class_probs[0].argmax(dim = 1) == class_index)]
sorted_indices = torch.argsort(class_detections[:, class_index], descending = True)
print(f"Full Probability Vector Max Value: {class_detections[sorted_indices][0].max(): .4f}")

img_tensor = img_tensor.detach()
model_output = model(img_tensor)
print(f"Model Output Max Sore: {model_output[0]['scores'][0]: .4f}")

>>> Full Probability Vector Max Value:  0.9851
>>> Model Output Max Sore:  0.9890
```

**Comment by AntoineSimoulin** [comment]
Hi @Aneesh-Sandhir, thanks for sharing your snippet! After reviewing it, I believe the differences you’re seeing stem from two main reasons:
1. Your code does not fully reproduce the forward pass of the ssdlite320_mobilenet_v3_large model. Specifically, some key operations are missing, such as the image [transform](https://github.com/pytorch/vision/blob/main/torchvision/models/detection/ssd.py#L354) and [postprocess](https://github.com/pytorch/vision/blob/main/torchvision/models/detection/ssd.py#L405) steps.
2. The variables you are comparing are not exactly the same. In your example, you check the probability of the first bounding box labeled "dog," but you compare it to the probability of the first bounding box returned by the model, which might correspond to a different label.

Addressing these points should help align your results with the model’s output.

**Comment by Aneesh-Sandhir** [comment]
Thank you @AntoineSimoulin for bringing the postprocess function to my attention. I had to augment said function top get the full probability vector for each of the detection output by the model.  

That being said I'm not sure I fully understand how the model preprocesses input images? The SSDlite documentation states that the images must be 320 x 320 with values scaled between 0 and 1 but I also see that ImageNet means and stanadrd deviations are used in the preprocessing of images which can lead to tensor values exceeding 1 and/or falling below 0. Are values the normalized values then rescaled between 0 and 1? Is it not actually an issue to have values in the input batch outside of the range specified in the documnetation?

**Comment by AntoineSimoulin** [comment]
@Aneesh-Sandhir 
* The [documentation](https://docs.pytorch.org/vision/main/models/generated/torchvision.models.detection.ssdlite320_mobilenet_v3_large.html#torchvision.models.detection.SSDLite320_MobileNet_V3_Large_Weights) you are referring to corresponds to the transformations used during the [training and evaluation](https://github.com/pytorch/vision/blob/main/references/detection/train.py#L50) of the model. In this case, we are using the [`ObjectDetection`](https://github.com/pytorch/vision/blob/main/torchvision/transforms/_presets.py#L23C7-L23C22) transforms, which indeed ensures image tensor inputs are within [0-1] range (as opposed to RGB value between [0-255] for instance). Typically the preprocessing outside of the model will apply transforms corresponding to `v2.Compose([v2.ToImage(), v2.ToDtype(torch.float32, scale=True)])`
* Within the model, we then apply normalization with the values defined in the [`ssdlite320_mobilenet_v3_large`](https://github.com/pytorch/vision/blob/main/torchvision/models/detection/ssdlite.py#L315) function. The normalization in the model will typically correspond to `v2.Normalize(mean=image_mean, std=image_std)`. Indeed this does not guarantees the image tensor values to be within [0-1] at this stage (the range should in fact be [[-1, 1]](https://github.com/pytorch/vision/blob/main/torchvision/models/detection/ssdlite.py#L314))

I hope this clarifies your additional question?

**Comment by Aneesh-Sandhir** [comment]
That makes sense, Thank you
